source/vir/src/prune.rs

```diff
@@ -341,6 +353,12 @@ fn traverse_typ(ctxt: &Ctxt, state: &mut State, t: &Typ) {
                 }
             }
         }
+        TypX::Opaque { .. } => {
+            // Revisit.
+            // For let x = foo<SomeType>(..);
+            // Do we need to traverse typ args (SomeType)? Probably not?
+            // All the type args should have been included in the function body through other means.
```

Comment:
prune.rs uses a Visitor and then calls `traverse_typ` for every node. So if you fix the visitor (like I suggested in another comment) there's no need to do recurse here.